src/bins.rs

```diff
@@ -146,3 +142,66 @@ struct Context<'c> {
 }
 
 impl<'c> Template for Context<'c> {}
+
+fn try_remove(path: &Path) {
+    debug!("Try to remove {}", path.display());
+    if let Err(err) = fs::remove_file(&path) {
+        debug!("Removing destination errored: {}", err);
+    }
+}
+
+fn install_move(src: &Path, dst: &Path) -> Result<(), BinstallError> {
+    debug!(
+        "Installing file (move) from '{}' to '{}'",
+        src.display(),
+        dst.display()
+    );
+
+    fs::rename(src, dst)?;
+    Ok(())
+}
+
+fn install_copy(src: &Path, dst: &Path) -> Result<(), BinstallError> {
+    debug!(
+        "Installing file (copy) from '{}' to '{}'",
+        src.display(),
+        dst.display()
+    );
+
+    #[cfg(feature = "reflink")]
+    {
+        debug!("Reflink copy or fallback");
+        reflink::reflink_or_copy(src, dst)?;
+    }
+    #[cfg(not(feature = "reflink"))]
+    {
+        debug!("Copy file");
+        fs::copy(src, dst)?;
+    }
+
+    debug!("Copy permissions");
+    let permissions = fs::File::open(src)?.metadata()?.permissions();
+    fs::File::open(dst)?.set_permissions(permissions)?;
+
+    Ok(())
+}
+
+fn install_link_or_copy(src: &Path, dst: &Path) -> Result<(), BinstallError> {
+    debug!(
+        "Installing file (link or copy) from '{}' to '{}'",
+        src.display(),
+        dst.display()
+    );
+
+    #[cfg(unix)]
+    {
```

Comment:
It's actually possible to also make symlink atomic.

Check the original `helpers::atomic_symlink` for implementation.